src/controllers/connectionManager.ts

```diff
@@ -722,6 +735,23 @@ export default class ConnectionManager {
         });
     }
 
+    /**
+	 * Gets the connection string to send to the middleware
+	 */
+    public async getConnectionString(connection: Interfaces.IConnectionCredentials): Promise<string> {
+		let connectionString: string;
+
+		if (connection.authenticationType === 'Integrated') {
+            connectionString =
+            `Data Source=${connection.server + (connection.port ? `,${connection.port}` : '')};Initial Catalog=${connection.database};Integrated Security=True`;
+		} else {
```

Comment:
Is this going to be correct if a port is specified? Don't you need to add the , in or is that already added in that case?

Reply:
I copied this code from ADS

Reply:
Doesn't mean it's correct. You should verify yourself to make sure it's doing the right thing because to me it seems like it isn't. 

---
In reply to: [327252219](https://github.com/microsoft/vscode-mssql/pull/1304#discussion_r327252219) [](ancestors = 327252219)